### ISSUE TITLE
[Bug]: Packed batch inference still falls back to per-sample execution

### ISSUE THREAD

**Issue by devUuung** [devUuung/CuKKS#1]
## Prerequisites
- [x] I have searched existing issues to avoid duplicates
- [x] I am using the latest version of CuKKS
- [x] I have read the [README](https://github.com/devUuung/CuKKS#readme)

## Bug Description
`encrypt_batch()` suggests a single-ciphertext SIMD inference path, but the current forward path still unpacks the packed ciphertext into per-sample encrypted tensors and runs the model sample-by-sample.

This matters because the public batch-processing example in `README.md` and `docs/concepts.md` reads like true packed inference, while the implementation currently routes packed inputs through `CKKSInferenceContext._forward_packed_batch()` and returns a Python `list[EncryptedTensor]` instead of preserving a packed `EncryptedTensor` end-to-end.

Relevant code paths:
- `cukks/nn/module.py`: `EncryptedModule.__call__()` detects `_packed_batch` and delegates to `context._forward_packed_batch()`
- `cukks/context.py`: `_forward_packed_batch()` isolates each sample and returns `[model(sample) for sample in enc_samples]`
- `.sisyphus/notepads/issue-4-fully-native-packed-batch/issues.md`: already documents the gap and the missing guardrails/tests

## Steps to Reproduce
```python
import torch
import cukks
from cukks.nn import EncryptedIdentity

ctx = cukks.CKKSInferenceContext(device="cpu")
samples = [torch.tensor([1.0, 2.0]), torch.tensor([3.0, 4.0])]
enc_batch = ctx.encrypt_batch(samples)

enc_output = EncryptedIdentity()(enc_batch)
print(type(enc_output))
print(isinstance(enc_output, list))
```

You can also see the current behavior directly in the implementation:
1. `EncryptedModule.__call__()` branches on `_packed_batch`
2. `CKKSInferenceContext._forward_packed_batch()` rotates/masks each sample
3. The method returns a Python list of per-sample outputs instead of a packed ciphertext

## Expected Behavior
Packed batch inference should stay in a packed representation through the forward path, or fail loudly if native packed execution is not yet supported. The current docs/examples imply a true SIMD forward path rather than a hidden per-sample fallback.

## Actual Behavior
Packed inputs are split into per-sample encrypted tensors and executed one-by-one. The output type becomes `list[EncryptedTensor]`, which hides the fact that the forward path is no longer operating as a single packed ciphertext.

## Environment
```yaml
- OS: Linux
- Python: 3.12
- CuKKS version: 0.1.2
- Branch: main (repository analysis on current HEAD)
- Backend: reproducible from source inspection; no GPU required to see the routing behavior
```

## Error Logs
```shell
No exception is raised. The issue is a silent fallback in the packed forward path.
```

## Additional Context
The docs currently advertise batch processing in a way that suggests end-to-end packed inference:
- `README.md` batch example uses `ctx.encrypt_batch(...)`, `enc_model(enc_batch)`, `ctx.decrypt_batch(...)`
- `docs/concepts.md` says packed samples allow all inferences to happen in parallel

That contract does not match the implementation today.
